.github/workflows/node.js.yaml

```diff
@@ -0,0 +1,35 @@
+name: Node.js CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "npm"
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: npm test
+      - name: Prettier check
+        uses: creyD/prettier_action@v4.2
+        with:
+          prettier_options: --check **/*.{ts,md}
+      - name: eslint check
+        run: |
+              # have to install eslint here for some reason... Other wise eslint wont be found
```

Comment:
I don't understand this, how are you testing?

Reply:
For some reason, running `npm run lint` locally works, but in the CI environment, it tells me it can't find eslint. I should do more testing, installing eslint in the action manually shouldn't be necessary I feel.

Reply:
That is bizarre.  If you want to file a follow-up bug in a new issue to look into this later, we can sort it out then.  It doesn't need to block this; but we shouldn't need to do it.

Reply:
Hmm, let me do some debugging. If I cant find anything in a reasonable time frame, I'll file an issue for later

Reply:
Found the issue. It was the prettier action doing some weird things making the `node_modules` vanish. Moving the eslint action above the prettier action fixed it.

Reply:
Hmm that would involve caching the node_modules manually? Let me look into how to do that

Reply:
The node action currently caches the node install it self, it does not cache node_modules

https://github.com/shawnyu5/starchart/actions/runs/4039795706/jobs/6944864665